File: 2021/1/1.rb
input = File.open("input").readlines.map(&:chomp).map(&:to_i)

# Take every value in overlapping pairs, and for each pair produce a 1 if increasing, 0 if
# decreasing, then add up the results
solution_1 = input.each_cons(2).map { |a, b| b > a ? 1 : 0 }.sum
puts "Problem 1: #{solution_1}"

# Take every triplet, add them up, then just do problem 1 again on those results
solution_2 = input.each_cons(3).map(&:sum).each_cons(2).map { |a, b| b > a ? 1 : 0 }.sum
puts "Problem 2: #{solution_2}"

File: 2022/4/1.py
file = open("input", "r")
input = list(map(lambda s: s.rstrip('\n'), file.readlines()))
file.close()

full_overlaps = 0
for line in input:
    pair = list(map(lambda s: list(map(lambda i: int(i), s.split("-"))), line.split(",")))
    if (pair[0][0] <= pair[1][0] and pair[0][1] >= pair[1][1] or
        pair[1][0] <= pair[0][0] and pair[1][1] >= pair[0][1]):
        full_overlaps += 1

print(full_overlaps)
